### DISCUSSION TITLE
Deploy "graphql-yoga" server with Vercel

### DISCUSSION THREAD

**Discussion by thibault60000** [vercel/vercel#4790] [Help]
Today I tried to deploy my project on Vercel. For the **Frontend Vue.JS** part it's ok.
But for the backend part **(graphql-yoga server / Apollo / Prisma)** I have a problem. When I try to access the url I get a **502 BAD GATEWAY** error. I created a vercel.json, .vercelignore and vercel.secrets.json

I think the problem comes from **the vercel.json** and the configuration of "build"

**I put the following code:**

```
{
  "version": 2,
  "builds": [
    {
      "src": "src/index.js",
      "use": "@now/node-server"
    }
  ],
  "routes": [
    {
      "src": "./*",
      "dest": "src/index.js"
    }
  ],
  "env": {
    "PRISMA_ENDPOINT": "@prisma_endpoint",
    "PRISMA_SECRET": "@prisma_secret",
    "APP_SECRET": "@app_secret"
  }
}
```

**My ./src/index.js file is as follows:**

```
const { GraphQLServer } = require("graphql-yoga");
const { Prisma } = require("prisma-binding");
const resolvers = require("./resolvers");

const server = new GraphQLServer({
  typeDefs: __dirname + "/schema.graphql",
  resolvers,
  context: (req) => ({
    ...req,
    db: new Prisma({
      typeDefs: "src/generated/prisma.graphql", 
      endpoint: process.env.PRISMA_ENDPOINT, 
      secret: process.env.PRISMA_SECRET, 
      debug: true, 
    }),
  }),
});

server.start(() => console.log("Server is running on http://localhost:4000"));
```

**In my log :** 

```
19:12:22.822  
Build completed. Populating build cache...
19:12:32.776  
Uploading build cache [33.45 MB]...
19:12:33.381  
Build cache uploaded: 604.787ms
19:12:33.393  
Done with "src/index.js"
```

THANKS 👍 
![2020-07-06_19h23_29](https://user-images.githubusercontent.com/18366294/86621298-3f9a7500-bfbe-11ea-9125-3e15cc301089.png)

**Comment by paulogdm** [comment]
Hello,

A few points:

1. `@now/node-server` is deprecated. Use `@vercel/node` instead.
2. We don't recommend that you use Vercel for API-only deployments. The platform is best suited for static and frontend deployments.
3. Check your logs for further information about what went wrong: https://vercel.com/blog/refined-logging
